### PR TITLE
Added Breadth First Search

### LINKED ISSUE #452: Adding Graph Traversal Algorithm Breadth First Search in C
# Description

I would like to add Graph Traversal Algorithm Breadth-First Search in C

# Additional context

I will be using the adjacency list data structure to do this

# Important Checklist 
-   [x] I have read the contribution guidelines before making this issue.
-   [x] I have verified this issue program is not already implemented
-   [x] I have not pushed this code anywhere else (even my repo)

# Languages and Assignees:
-   [x] C: @Sashi445

### PR DESCRIPTION
**Pull request by Sashi445**
# Description

Added Breadth First Search Algorithm for Adjacency List Data Structure!!

Fixes #452 

## Type of change

New Feature

**Please delete options that are not relevant**

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

# Checklist:

-   [x] I have read the contribution guidelines before making this PR.
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented in my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation (Language folder or any other readme instructed to)
-   [x] My changes generates no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kaustubhgupta** [comment]
Fill the PR template properly. I am not able to identify whether you were assigned any issue or not

**Comment by Sashi445** [comment]
> Fill the PR template properly. I am not able to identify whether you were assigned any issue or not

I've labeled it and added the issue number which I was assigned. Thank You!!

File: C/BFS_AdjacencyList.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+void addEdge(adjList* list, int destination, int weight){
+	node* temp = newNode(rand() % destination, weight);
+	temp->next = list->edgeList;
+	list->edgeList = temp;
+}
+
+// creates a random weighted graph(can be customised to make a graph with specific requirements)
+//createGraph() takes in number of nodes as parameter and returns a randomly generated weighted graph
+
+Graph* createGraph(int n){
```
Comment (YashSaxena9) [inline C/BFS_AdjacencyList.c:91]: Nice workaround for random graph

File: C/BFS_AdjacencyList.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+}adjList; // a structure representing the start vertex and all of its neighbours(edges)
+
+typedef struct graph{
+	int n;
+	adjList** vertices;
+}Graph; // a structure representing whole of the graph DS(adjacency list)
+
+// Queue Implementation (Required for breadth first search)
+
+typedef struct queue{
```
Comment (YashSaxena9) [inline C/BFS_AdjacencyList.c:26]: Self-implemented queue is appreciable, you can also use an array-based queue for faster performance :)

File: C/BFS_AdjacencyList.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+typedef struct node{
+	int dest;
+	int weight;
+	struct node* next;
+}node; // a struture to denote vertices in a edgelist
+
+typedef struct adjList{
+	int startIndex;
+	int visited;
```
Comment (YashSaxena9) [inline C/BFS_AdjacencyList.c:15]: Overall logic of the code is correct but correct choice of data type and data structure for implementation is suggested. It increases code readability and lowers time and space complexity :)

**Review by YashSaxena9 — APPROVED** [review]

**Comment by YashSaxena9** [comment]
PR accepted, thank you for your hard work :)

**Pull request merged**